src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,117 @@
+
+  /**
+   * Set up state for handling profile creation-related requests. This method is only called when
+   * running in a server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setProfileStore(ProfileStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setUserStore(UserStore.getInstance());
+  }
+
+  /**
+   * Sets the ProfileStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setProfileStore(ProfileStore profileStore) {
+    this.profileStore = profileStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  /** doGet function will eventually output HTML directly from servlet */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    String requestUrl = request.getRequestURI();
+    String username = requestUrl.substring("/user/".length());
+    User user = userStore.getUser(username);
+
+    Profile profile = profileStore.getProfile(username);
+    if (profile == null) {
+      // couldn't find profile, redirect to homepage
+      System.out.println("Profile was null: " + username);
+      response.sendRedirect("/index.jsp");
+      return;
+    }
+    List<Message> messages = messageStore.getUserMessages(user.getId());
+
+    request.setAttribute("profileUser", user);
+    request.setAttribute("username", username);
+    request.setAttribute("profile", profile);
+    request.setAttribute("about", profile.getAbout());
+    request.setAttribute("messages", messages);
+    request.setAttribute("photo", profile.getPhoto());
+
+    request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    String requestUrl = request.getRequestURI();
+    String username = requestUrl.substring("/user/".length());
+    User user = userStore.getUser(username);
+
+    Profile profile = profileStore.getProfile(username);
+    if (profile == null) {
+      // couldn't find profile, redirect to homepage
+      response.sendRedirect("/index.jsp");
+      return;
+    }
+    if (request.getParameter("update") != null) {
+      String newAbout = request.getParameter("newAbout");
+      profile.setAbout(newAbout);
```

Comment:
@sgj1 An example here on how to update the ProfileStore when updating a Profile. Similarly for after password change, you'd need to update the User object in the UserStore. Also see changes in the profileStore